src/ApparentHorizons/ComputeExcisionBoundaryVolumeQuantities.tpp

```diff
@@ -0,0 +1,221 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ApparentHorizons/ComputeExcisionBoundaryVolumeQuantities.hpp"
+
+#pragma once
+
+#include <cstdint>
+#include <type_traits>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tags/TempTensor.hpp"
+#include "DataStructures/TempBuffer.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Christoffel.hpp"
+#include "PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/Christoffel.hpp"
+#include "PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/ExtrinsicCurvature.hpp"
+#include "PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/Ricci.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Lapse.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Shift.hpp"
+#include "PointwiseFunctions/GeneralRelativity/SpacetimeNormalVector.hpp"
+#include "PointwiseFunctions/GeneralRelativity/SpatialMetric.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Transform.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace ah {
+
+/// Single frame case
+template <typename SrcTagList, typename DestTagList>
+void ComputeExcisionBoundaryVolumeQuantities::apply(
+    const gsl::not_null<Variables<DestTagList>*> target_vars,
+    const Variables<SrcTagList>& src_vars, const Mesh<3>& /*mesh*/) {
+  static_assert(
+      std::is_same_v<tmpl::list_difference<SrcTagList, allowed_src_tags>,
+                     tmpl::list<>>,
+      "Found a src tag that is not allowed");
+  static_assert(
+      std::is_same_v<tmpl::list_difference<required_src_tags, SrcTagList>,
+                     tmpl::list<>>,
+      "A required src tag is missing");
+
+  static_assert(
+      std::is_same_v<tmpl::list_difference<DestTagList,
+                                           allowed_dest_tags<Frame::Inertial>>,
+                     tmpl::list<>>,
+      "Found a dest tag that is not allowed");
+  static_assert(
+      std::is_same_v<tmpl::list_difference<required_dest_tags<Frame::Inertial>,
+                                           DestTagList>,
+                     tmpl::list<>>,
+      "A required dest tag is missing");
+
+  if (target_vars->number_of_grid_points() !=
+      src_vars.number_of_grid_points()) {
+    target_vars->initialize(src_vars.number_of_grid_points());
+  }
+
+  get<gr::Tags::SpacetimeMetric<3, Frame::Inertial>>(*target_vars) =
+    get<gr::Tags::SpacetimeMetric<3, Frame::Inertial>>(src_vars);
```

Comment:
This whole class is extremely complicated when all it does is forward the spacetime metric. Are we going to put more logic here? Can you reuse the Apparent horizon class instead? Why are the two apply functions so different?

Reply:
1. Yes. We need to compute all ingredients for char speeds on a moving mesh.
2. No, because AH uses different variables.
3. One for static meshes and one for moving meshes.

Reply:
there's some commented-out code, so some cleanup is needed here I think